src/main/java/com/exasol/parquetio/splitter/ParquetFileSplitter.java

```diff
@@ -0,0 +1,79 @@
+package com.exasol.parquetio.splitter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.parquetio.data.ChunkInterval;
+import com.exasol.parquetio.data.Interval;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.io.InputFile;
+
+/**
+ * A class that splits Parquet file into chunks of certain size.
+ *
+ * Each chunk then contains one or more row group start and end positions in a Parquet file.
+ */
+public class ParquetFileSplitter implements FileSplitter {
+
+    private static final long DEFAULT_CHUNK_SIZE = 64L * 1024 * 1024;
+    private final InputFile file;
+    private final long chunkSize;
+
+    /**
+     * Creates a new instance of {@link ParquetFileSplitter}.
+     *
+     * It uses default chunk size of {@code 64MB} to split the file.
+     *
+     * @param file a Parquet file
+     */
+    public ParquetFileSplitter(final InputFile file) {
+        this(file, DEFAULT_CHUNK_SIZE);
+    }
+
+    /**
+     * Creates a new instance of {@link ParquetFileSplitter}.
+     *
+     * @param file a Parquet file
+     * @param chunkSize a chunk size in bytes
+     */
+    public ParquetFileSplitter(final InputFile file, final long chunkSize) {
+        this.file = file;
+        this.chunkSize = chunkSize;
+    }
+
+    @Override
+    public List<Interval> getSplits() {
+        try (final var reader = ParquetFileReader.open(file)) {
+            return getRowGroupSplits(reader.getRowGroups());
+        } catch (IOException exception) {
+            throw new IllegalStateException(ExaError
+                .messageBuilder("E-PIOJ-4")
+                .message("Failed to open a Parquet file {{FILE}} for splitting.", this.file.toString()).toString(),
+                exception
+            );
+        }
+    }
+
+    protected List<Interval> getRowGroupSplits(final List<BlockMetaData> rowGroups) {
+        final List<Interval> chunks = new ArrayList<>();
+        final long end = rowGroups.size();
+        var startPosition = 0L;
+        var currentSize = 0L;
+        for (var endPosition = 0; endPosition < end; endPosition++) {
+            currentSize += rowGroups.get(endPosition).getTotalByteSize();
+            if (currentSize >= this.chunkSize) {
+                chunks.add(new ChunkInterval(startPosition, endPosition + 1L));
+                startPosition = endPosition + 1L;
+                currentSize = 0L;
+            }
+        }
```

Comment:
This operation looks expensive. Have you checked how long the planning takes? Maybe assuming that all row groups are equally sized would be an acceptable trade off.

Reply:
Or we could split the file size into partitions and determine which block belong into a partition during reading.

Reply:
No it is not expensive. Number of row groups, their sizes are already written with file at the footer. It only reads the available metadata from file itself.

They are mostly equally sized, by default ~128MB, but users can set while writing using `.withRowGroupSize(rowGroupSize)`.